From bddaaf5980d0a70f8f7b7906b0bf4cfb0a05c7c2 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.i.king@gmail.com>
Date: Thu, 5 Dec 2024 13:07:10 +0000
Subject: [PATCH] src: thd_dbus_interface: add extra sanity check on
 introspection_data
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

gcc static analyzer warns that introspection_data is not being null checked
which could lead to a potential null pointer dereference. Even though error
is being checked, it seems prudent to add a null check on introspection_data
to make the code more resiliant and also to silence static analysis warnings.

Signed-off-by: Colin Ian King <colin.i.king@gmail.com>
---
 src/thd_dbus_interface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/thd_dbus_interface.cpp b/src/thd_dbus_interface.cpp
index c75e56c..6ae7af1 100644
--- a/src/thd_dbus_interface.cpp
+++ b/src/thd_dbus_interface.cpp
@@ -1172,7 +1172,7 @@ thd_dbus_on_bus_acquired(GDBusConnection *connection,
 
 	introspection_data = thd_dbus_load_introspection("src/thd_dbus_interface.xml",
 							 &error);
-	if (error != NULL) {
+	if (introspection_data == NULL || error != NULL) {
 		thd_log_error("Couldn't create introspection data: %s:\n",
 			      error->message);
 		return;
-- 
2.47.2

